openmp/runtime/src/ompd-specific.cpp

```diff
@@ -78,9 +78,16 @@ void ompd_init()
   OMPD_FOREACH_SIZEOF(ompd_init_sizeof)
 #undef ompd_init_sizeof
 
-  volatile static const char * ompd_my_dll_locations[2] = {"libompd.so",NULL};
-  
-
+  // Find the location of libomp.so thru dladdr and replace the libomp with libompd to get
+  // the full path of libompd
+  Dl_info dl_info;
+  int ret = dladdr((void*)ompd_init, &dl_info);
+  if (!ret) {
+	fprintf(stderr, "%s\n", dlerror());
+  }
+  char* libname = (char*)malloc (strlen(dl_info.dli_fname)+ 2/*for 'd' and '\0'*/);
+  strcpy (libname, dl_info.dli_fname);
+  memcpy(strrchr(libname, '/'), "/libompd.so\0",12);
```

Comment:
As I understand this code, the assumption is that the library is called "libomp.so"?
If the library name is longer, there should not be a problem. If the library name is shorter, we would have a buffer overflow. Should we get the length like: `strrchr(dl_info.dli_fname, '/') - dl_info.dli_fname`, probably make sure that strrchr does not return NULL.

Also, I'm not sure how much we care about Windows compatibility at this point. I think, that the libomp part of OMPD should currently build on Windows. With dl* this changes. Probably wrap this part of the code in `#if KMP_OS_UNIX` and use `char* libname = "";` by default.

Reply:
Added "UNX GUARD" and modified calculation of string length as suggested.